src/System.Configuration/src/System/Configuration/IdentityHelper.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace System.Configuration
+{
+    internal static class IdentityHelper
+    {
+        private static readonly char[] s_base32Char =
+        {
+            'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h',
+            'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p',
+            'q', 'r', 's', 't', 'u', 'v', 'w', 'x',
+            'y', 'z', '0', '1', '2', '3', '4', '5'
+        };
+
+        internal static string GetNormalizedUriHash(Uri uri)
+        {
+            // On desktop System.Security.Url is used as evidence, it has an internal Normalize() method.
+            // Uri.ToString() appears to be functionally equivalent.
+            return GetStrongHashSuitableForObjectName(uri.ToString());
+        }
+
+        internal static string GetNormalizedStrongNameHash(AssemblyName name)
+        {
+            byte[] publicKey = name.GetPublicKey();
+
+            // If we don't have a key, we're not strong named
+            if (publicKey == null || publicKey.Length == 0)
+                return null;
+
+            // Emulate what we get from StrongName.Normalize().
+            using (MemoryStream ms = new MemoryStream())
+            {
+                using (BinaryWriter bw = new BinaryWriter(ms))
+                {
+                    bw.Write(publicKey);
+                    bw.Write(name.Version.Major);
+                    bw.Write(name.Name);
+
+                    ms.Position = 0;
+                    return GetStrongHashSuitableForObjectName(ms);
+                }
+            }
+        }
+
+        internal static string GetStrongHashSuitableForObjectName(string name)
+        {
+            using (MemoryStream ms = new MemoryStream())
+            {
+                using (BinaryWriter b = new BinaryWriter(ms))
```

Comment:
nit: it's shorter to use one set of {}

            using (MemoryStream ms = new MemoryStream())
            using (BinaryWriter b = new BinaryWriter(ms))